glDemo/main.cpp

```diff
@@ -86,6 +100,41 @@ int main() {
 	// *** setup viewplane to the appropriate size
 	gluOrtho2D(-1.1f, 1.1f, -1.1f, 1.1f);
 
+
+	random_device rd;
+	engine = mt19937(rd());
+	range = uniform_real_distribution<float>(-1.0f, 1.0f);
+
+	random_device dr;
+	engine2 = mt19937(dr());
+	range2 = uniform_real_distribution<float>(-1.0f, 20.0f);
+
+	// Create a openGL vec2 vector object containing 100 elements all initialised to 0.0
+	vertexCoords = vector<glm::vec2>(100, glm::vec2(0.0f, 0.0f));
+	colourCodes = vector<glm::vec4>(100, glm::vec4(0.0f, 0.0f, 0.0f, 0.0f));
+	sizeValue = vector<float>(100, 0.0f);
+
+
+	// Iterate 100 times (once for each vertex) and calculate and store the random x and y values
+	for (int i = 0; i < 100; i++) {
+
+		float x = range(engine);
+		float y = range(engine);
+
+		vertexCoords[i] = glm::vec2(x, y);
+
+		float r = range(engine);
```

Comment:
In the colour setup code you use the same range as the position data, but this goes from -1 to +1.  Ideally for the colour data you'd have a range from 0 to 1